app/models/spree/line_item.rb

```diff
@@ -234,10 +234,13 @@ def unit_value
       final_weight_volume / quantity
     end
 
-    def unit_price_price_and_unit
-      unit_price = UnitPrice.new(variant)
-      Spree::Money.new(price_with_adjustments / unit_price.denominator).to_html +
-        "&nbsp;/&nbsp;".html_safe + unit_price.unit
+    def unit_price
+      unit = UnitPrice.new(variant).unit
+      amount = price_with_adjustments / UnitPrice.new(variant).denominator
+      {
+        amount:,
+        unit:,
+      }
     end
```

Comment:
It's almost not worth bringing up, but I was wondering why UnitPrice was instantiated twice. It may be a bit clearer to do that only once. And then we can do the whole thing a bit shorter. What do you think about this version?

```suggestion
    def unit_price
      unit_price = UnitPrice.new(variant)
      {
        amount: price_with_adjustments / unit_price.denominator,
        unit: unit_price.unit,
      }
    end
```